Source/WTF/wtf/NativePromise.h

```diff
@@ -940,7 +941,7 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
         {
             ASSERT(m_thenCallback, "Can only track a request once");
             requestHolder.track(*m_thenCallback);
-            m_promise->maybeSettle(m_thenCallback.releaseNonNull(), m_logSiteIdentifier);
+            protectedPromise()->maybeSettle(m_thenCallback.releaseNonNull(), m_logSiteIdentifier);
```

Comment:
There are no static analyzer warnings about `m_promise` or `m_thenCallback` in NativePromise.h. Could you explain why we're making these changes?

Reply:
Ah it seems like this is a templated class and so we need to look at the failures in WebKitBuild/. So we do want these changes.